js/example_roulette.js

```diff
@@ -0,0 +1,116 @@
+    if (typeof args !== "undefined") { args = args.innerText; }
+
+    // check whether stdin has been defined (RUNNABLE_EXAMPLE_STDIN macro in Ddoc)
+    var stdin = findFirst(parent.children, function(e){ return e.className === "runnable-examples-stdin";});
+    if (typeof stdin !== "undefined") { stdin = stdin.innerText; }
+
+    // wraps the <pre> code box in a <div> element
+    // this rewrapping is needed for the editor framework
+    var dCode = document.createElement("div");
+    dCode.className = "d_code";
+    dCode.append(pre);
+
+    // create the CodeMirror editor overlay
+    var el = document.createElement("div");
+    el.style.display = "block";
+    el.className = "d_run_code";
+    // code box (CodeMirror will be loaded into this element)
+    var dCodeText = document.createElement("textarea");
+    dCodeText.className = "d_code";
+    dCodeText.style.display = "none";
+    el.appendChild(dCodeText);
+
+    // create optional stdin overlay
+    if (stdin) { el.appendChild(createOverlay("d_code_stdin", "Input", "textarea", stdin)); }
+
+    // create optional args overlay
+    if (args) { el.appendChild(createOverlay("d_code_stdin", "Command line arguments", "textarea", args)); }
+
+    // create main output overlay
+    el.appendChild(createOverlay("d_code_output", "Application output", "pre", "Running..."));
+
+    // user interaction buttons
+    el.appendChild(createButton("editButton", "Edit"));
+    if (stdin) { el.appendChild(createButton("inputButton", "Input")); }
+    if (args) { el.appendChild(createButton("argsButton", "Args")); }
+    el.appendChild(createButton("runButton", "Run"));
+    el.appendChild(createButton("resetButton", "Reset"));
+
+    // replace the <pre> element with its enhanced wrapper div
+    var root = document.createElement("div");
+    root.appendChild(dCode);
+    root.appendChild(el);
+    parent.appendChild(root);
+    return root;
+}
+/**
+Searches any object for the first hit.
+Params:
+    el = object to filter
+    fn = function to be applied as filter
+Returns: first hit, `undefined` otherwise
+*/
+function findFirst(el, fn) {
+    return Array.prototype.filter.call(el, fn)[0];
+}
+
+(function() {
+// randomly pick one example and bootstrap the runnable editor
+var examples = document.getElementsByClassName('your-code-here-extra');
+var rouletteIndex = Math.floor(Math.random() * examples.length);
+var rouletteChild = examples[rouletteIndex];
+
+// step 1: select the .runnable-example <div>
+var rouletteChildNode = findFirst(rouletteChild.children, function(e) { return e.className.indexOf("runnable-example") != -1; });
+// step 2: select the first <pre> element (of the .runnable-example <div>)
+var el = findFirst(rouletteChildNode.children, function(e) { return e.nodeName == "PRE";});
+// step 3: Construct runnable example layout within the <pre> element
+buildRunnableExampleLayout(el);
```

Comment:
@CyberShadow 
> Also I still have a strong suspicion that the second one could be much simpler as mentioned above.

Yes, these lines without `buildRunnableExampleLayout` would - in theory - be enough. However, it still "jumps" due to the buttons added and just
- adding the buttons isn't a good solution imho, because (1) it creates code entanglement (at another random, JS place the script needs to be aware of this) and (2) it creates code redundancy and makes it a lot harder if someone wants to change the overall layout
- adding sth. `el.style.marginBottom = "20px"` is a similar hack (and still flashes/"jumps")